### DISCUSSION TITLE
Netherlands: Zonneplan (2025)

### DISCUSSION THREAD

**Discussion by tieskuh** [JaccoR/hass-entso-e#103] [Price Modifyer Templates]
~~`{% set s = {"handling_fee": 0.0024793, "energie_belasting": 0.12599} %} {{(current_price + s.handling_fee + s.energie_belasting ) | float}}`~~

~~Update 1-7-2023:
`{% set s = {"handling_fee": 0.01653, "energie_belasting": 0.12599} %} {{(current_price + s.handling_fee + s.energie_belasting ) | float}}`~~

~~Update 1-1-2024:
`{% set s = {"handling_fee": 0.01653, "energie_belasting": 0.10880} %} {{(current_price + s.handling_fee + s.energie_belasting ) | float}}`~~

Update 1-1-2025:
`{% set s = {"handling_fee": 0.01653, "energie_belasting": 0.10154} %} {{(current_price + s.handling_fee + s.energie_belasting ) | float}}`

And set VAT to 0.21

<img width="328" alt="image" src="https://user-images.githubusercontent.com/115901851/210183457-24b8119d-4e07-402b-b7b9-63f6d52fc43f.png">

ENTSO-e sensor:
![Afbeelding van WhatsApp op 2023-01-01 om 21 08 20](https://user-images.githubusercontent.com/115901851/210183459-d1ef0510-5b75-43af-ba11-4e88064c4a90.jpg)

Sensor from the Zonneplan integration (https://github.com/fsaris/home-assistant-zonneplan-one) at the same moment:
![Afbeelding van WhatsApp op 2023-01-01 om 21 08 20](https://user-images.githubusercontent.com/115901851/210183461-6036ec36-8c7c-445a-ba0c-78f651909478.jpg)

**Comment by thierrybla** [comment]
Moet VAT niet naar 9%?

**Reply by tieskuh** [reply]
Was het maar zo'n feest...

https://www.rijksoverheid.nl/actueel/nieuws/2022/12/22/ministerie-van-economische-zaken-en-klimaat-belangrijkste-veranderingen-per-1-januari-2023

"Met de komst van dit prijsplafond vervalt per 1 januari 2023 de btw-verlaging op energie."

**Comment by YarneThatsMe** [comment]
Sinds 1 juli zijn de kosten bij zonneplan veranderd, is het mogelijk dat bovenstaande formule aangepast kan worden?

**Reply by tieskuh** [reply]
Volgens mij zijn alleen de handlingkosten veranderd. De nieuwe template zou dan dit moeten worden:

 {% set s = {"handling_fee": 0.02, "energie_belasting": 0.12599} %} {{(current_price + s.handling_fee + s.energie_belasting ) | float}}

**Reply by YarneThatsMe** [reply]
Ik denk dat het toch nog niet helemaal klopt, als ik in de zonneplan app kijk staat er bij het duurste uur 37,92 cent en in hass met de nieuwe formule 36,22. Volgens mij is niet alleen de handling fee veranderd, of ik zie iets over het hoofd

**Reply by tieskuh** [reply]
Ik zie wat je bedoelt. Bij mij komt het ook niet helemaal overeen. Vreemd want dit is precies wat Zonneplan communiceert over het tarief:

<img width="619" alt="image" src="https://github.com/JaccoR/hass-entso-e/assets/115901851/58f5b865-d838-4a0b-ad75-58d5dd2695d2">

kWh-prijs + handling fee € 0,02 + energiebelasting € 0,12599
Over het geheel betaal je 21% btw.

Ik begrijp ook niet helemaal waar het prijsverschil vandaan komt. Ik zal er later nog eens naar kijken.

**Comment by YarneThatsMe** [comment]
Als ik mijn berekening handmatig doe, dan klopt het wel. Ik zag dat de inkoopprijs voor een kWh 0,09 was, dus de berekening is als volgt:
Inkoopprijs 0,09+ Energiebelasting 0,12599=0,21599 0,21599+(0,21599*BTW 0,21)+Handling fee 0,02=0,2813
Dit is wat zonneplan ook aangaf in de app.

De handling fee gaat dus niet mee in de BTW. Als ik de handling fee inclusief de BTW doe krijg ik de volgende berekening:
0,21599+0,02=0,23599 0,23599+(0,23599*0,21)=0,2855479
Home Assistant gaf 0,28552 aan, komt dus bijna overeen...

Omdat de handling fee dus eerst zo klein was zag je dat niet terug in de prijs. Omdat de fee dus nu zo aanzienlijk verhoogd is komt dat wel terug in de prijs. Ik weet niet of de entso-e integratie of de som aangepast kan worden dat het BTW tarief niet over de handling fee wordt berekend.

Of klopt mijn berekening niet/zie ik wat over het hoofd?

**Reply by tieskuh** [reply]
Ah, dat zal het verklaren. Bedankt voor het uitzoeken. Er zit wel btw op de handling fee volgens mij maar het bedrag dat ze noemen zal wel inclusief btw zijn.

**Reply by tieskuh** [reply]
Als je wat verder doorklikt op de website staat het er uitgelegd.

<img width="528" alt="image" src="https://github.com/JaccoR/hass-entso-e/assets/115901851/c3edcb8a-8ca9-4398-86d6-e490e1cb3f81">

**Comment by YarneThatsMe** [comment]
Als je in de integratie het BTW tarief op 0 zet en deze in de calculatie meeneemt is het voor nu gefixt:
{% set s = {"btw": 0.21,"handling_fee": 0.02, "energie_belasting": 0.12599} %} {{(current_price * s.btw + s.energie_belasting * s.btw + current_price + s.energie_belasting + s.handling_fee ) | float}}

Nu klopt de berekening weer. Ik ben alleen niet thuis in Jinja2 templating, dus ik weet even niet hoe je netjes de berekening kan maken, maar deze werkt wel!

**Reply by tieskuh** [reply]
Ik stel dit voor: 

`{% set s = {"handling_fee": 0.01653, "energie_belasting": 0.12599} %} {{(current_price + s.handling_fee + s.energie_belasting ) | float}}`

(en dan de btw op 0.21 laten staan)

**Comment by steef12111** [comment]
Hi, ik probeer mijn prijzen te matchen met de site van zonneplan, maar dat lijkt me niet te lukken. 
Ik gebruik het volgende:
{% set s = {"handling_fee": 0.02, "energie_belasting": 0.13165} %} {{(current_price + s.handling_fee + s.energie_belasting ) | float}}

De prijzen die ik in Home Assistant zie zijn hoger dan de prijzen van Zonneplan zelf.
Iemand enig idee hoe dit zou kunnen?

**Reply by tieskuh** [reply]
Als je de VAT op 0.21 laat staan, dan zijn dit de juiste prijzen:

```{% set s = {"handling_fee": 0.01653, "energie_belasting": 0.10880} %} {{(current_price + s.handling_fee + s.energie_belasting ) | float}}```

![image](https://github.com/JaccoR/hass-entso-e/assets/115901851/61ccdcd8-dea0-4b08-bc86-5048271f7813)

De prijzen die jij invoert zijn inclusief btw. Dat kan natuurlijk ook, maar dan moet je VAT op 0 zetten.

**Reply by steef12111** [reply]
Ah check, bedankt voor snelle response. Wanneer ik dit invoer zit er nog een verschil tussen de prijzen in HA en de site van Zonneplan. Of ontbreekt er op de site van Zonneplan een bepaalde fee?
![Zonneplan_HA](https://github.com/JaccoR/hass-entso-e/assets/46138546/74bfe006-b7aa-445f-b45b-918a3d34c122)
![Zonneplan_HA2](https://github.com/JaccoR/hass-entso-e/assets/46138546/fb3729b3-8c63-4754-b65d-c90c4d2ac019)

**Reply by tieskuh** [reply]
Ik denk dat je even moet wachten op de volgende update van de prijzen. Volgens mij update de entso-e integratie de prijzen pas bij de volgende update. Als je de price modifier template nu goed hebt staan, dan vermoed ik dat de prijzen morgen wel goed zijn als de prijzen voor zondag beschikbaar zijn.

Andere optie is de Zonneplan integratie gebruiken: https://github.com/fsaris/home-assistant-zonneplan-one
Die haalt de prijs uit de Zonneplan-app.

**Reply by steef12111** [reply]
Hi, weer even verder. Op het oog leek mijn entso-e integratie correcte waardes te produceren, totdat ik vandaag de zonneplan integratie ernaast hield. Het lijkt erop dat wanneer de energieprijs negatief is, dit niet goed verwerkt wordt.
Ben ik de enige met dit probleem?

Boven is de entso-e integratie, onder is de zonneplan-one integratie
![dynamic_morgen](https://github.com/JaccoR/hass-entso-e/assets/46138546/8cc67c41-fec2-462c-a6f2-8e68f1601971)

Boven is de entso-e integratie, onder is de zonneplan-one integratie
![dynamic_vandaag](https://github.com/JaccoR/hass-entso-e/assets/46138546/8be1c700-5497-488e-8646-8d256ee01b2e)

**Reply by AbelLykens** [reply]
Heeft wellicht te maken met: https://www.icis.com/explore/resources/news/2024/06/25/11011378/day-ahead-power-price-spreads-soar-as-markets-decouple-on-technical-issue

**Comment by fvdm** [comment]
Per 01-01-2026 is de energiebelasting €0,09161/kWh. Zonneplan toeslag is onveranderd en BTW blijft 21%.

{% set s = {"handling_fee": 0.01653, "energie_belasting": 0.09161} %} {{(current_price + s.handling_fee + s.energie_belasting) | float}}

**Comment by Kurk-1988** [comment]
de van FVDM van 27 dagen geleden lijkt te kloppen, wijkt een klein beetje af maar dat zal wel in de afronding van getallen zitten in de app van zonneplan en de HA integratie

**Comment by fvdm** [comment]
@Kurk-1988 Dat zit hem inderdaad in afronding, maar waar precies? Als ik de tarieven hanteer van de klantenservice dan krijg ik ook iets anders dan wat de API meldt. Maar als ik exact terugreken van wat de API rapporteert dan krijg een iets andere handling fee: `0.016538781`

<details>
<summary>Verschillende rekenmethodes...</summary>

#### Kale [marktprijs](https://transparency.entsoe.eu/market/energyPrices?appState=%7B%22sa%22%3A%5B%22BZN%7C10YNL----------L%22%5D%2C%22st%22%3A%22BZN%22%2C%22mm%22%3Atrue%2C%22ma%22%3Afalse%2C%22sp%22%3A%22HALF%22%2C%22dt%22%3A%22TABLE%22%2C%22df%22%3A%222026-01-28%22%2C%22tz%22%3A%22CET%22%7D)

Moment: 2026-01-28 09:00

label         | tarief
:-------------|:------
9:00          | 182.73
9:15          | 180.12
9:30          | 167.78
9:45          | 164.60
Gemiddeld 60M | 173.8075
Per kWh       | 0.1738075


### Terugrekenen naar handling fee

label          | tarief
:--------------|:------
ZP API met BTW | 0.341155
Zonder BTW     | 0.281946281
Markt 60M      | 0.1738075
[EB <10kWh](https://www.belastingdienst.nl/wps/wcm/connect/bldcontentnl/belastingdienst/zakelijk/overige_belastingen/belastingen_op_milieugrondslag/energiebelasting/)      | 0.0916
Handling fee   | **0.016538781**

Dan is de handling dus `0.02001192501` en niet `0.02` inclusief BTW.


### Wat het zou moeten zijn

label            | tarief
:----------------|:------
Markt 60M        | 0.1738075
Energiebelasting | 0.0916
Tarief excl. BTW | 0.2654075
Tarief incl. BTW | 0.321143075
Handling fee     | 0.02
Netto tarief     | **0.341143075**

Maar met afronding wordt dat dan `0.3411431` dus waar die `0.341155` van de API vandaan komt is mij een raadsel.
</details>

### tl;dr

{% set s = {"handling_fee": 0.016538781, "energie_belasting": 0.0916} %} {{(current_price + s.handling_fee + s.energie_belasting) | float}}

_En de ENTSO-e integratie rond af op 5 decimalen en Zonneplan op 6._
